datanode/dehistory/store/store.go

```diff
@@ -686,7 +695,10 @@ func createIpfsNodeConfiguration(log *logging.Logger, identity config.Identity,
 
 	updatedSwarmAddrs := make([]string, 0, 10)
 	for _, addr := range cfg.Addresses.Swarm {
-		updatedSwarmAddrs = append(updatedSwarmAddrs, strings.ReplaceAll(addr, "4001", strconv.Itoa(swarmPort)))
+		// Exclude ip6 addresses cause hang on lookup
+		if !strings.Contains(addr, "/ip6/") {
+			updatedSwarmAddrs = append(updatedSwarmAddrs, strings.ReplaceAll(addr, "4001", strconv.Itoa(swarmPort)))
```

Comment:
Why are we replacing all 4001 here? would be good to have a comment to explain here, it just seems like out of nowhere.

Reply:
ah yeah, instroduced a named const to make it clearer